apps/cowswap-frontend/src/modules/tradeQuote/utils/quoteUsingSameParameters.ts

```diff
@@ -14,28 +14,35 @@ import { AppDataInfo } from 'modules/appData'
 export function quoteUsingSameParameters(
   chainId: SupportedChainId,
   currentParams: Nullish<TradeParameters>,
-  nextParams: Nullish<TradeParameters>,
+  nextParams: Nullish<QuoteBridgeRequest>,
   currentAppData: AppDataInfo['doc'] | undefined,
   appData: AppDataInfo['doc'] | undefined,
 ): boolean {
   if (!currentParams || !nextParams) return false
 
+  /**
+   * TODO: in bridging mode buy token changes to intermediate token
+   * TODO: in bridging mode, appData contains only hooks without other information
+   * TODO: in bridging mode, receiver became a CoW Shed address
+   */
+  const isBridge = true
```

Comment:
This is a bit of hell...
This function is responsible for skiping quote fetching when nothing real is changed.
With briding quotes it's a bit different than with CoW Protocol quotes.
1. AppData comparation doesn't work. Probably it's related to https://github.com/cowprotocol/cow-sdk/pull/275/files#r2024716345. Need to check why it's different
2. Bridge quote changes buyToken to intermediate one. As I mentioned [here](https://github.com/cowprotocol/cowswap/pull/5588/files#r2024702243), we need to save `BridgeQuoteAndPost['bridge']` into `tradeQuotesAtom` and use the state here in order to properly compare buy tokens.
3. Bridge quote changes to CoW Shed, we also need to handle that fact here